plugin-server/src/main/ingestion-queues/batch-processing/each-batch-ingestion.ts

```diff
@@ -255,7 +255,7 @@ async function emitToOverflow(queue: IngestionConsumer, kafkaMessages: Message[]
             queue.pluginsServer.kafkaProducer.produce({
                 topic: KAFKA_EVENTS_PLUGIN_INGESTION_OVERFLOW,
                 value: message.value,
-                key: null, // No locality guarantees in overflow
+                key: message.key,
```

Comment:
I think it's better to have this configurable in `PluginServerConfig`, to be able to remove locality during an incident without waiting for a code deploy.

Reply:
Makes sense, I added the `INGESTION_OVERFLOW_PRESERVE_PARTITION_LOCALITY` to handle this: c1ad17ec6efa48347aedf769f736911109185d53